### PR TITLE
[target-allocator] create new target package

### PR DESCRIPTION
**Pull request by moh-osman3**
Resolves https://github.com/open-telemetry/opentelemetry-operator/issues/1213

### CODE REVIEW

File: cmd/otel-allocator/allocation/consistent_hashing.go [hunk 1]
```diff
@@ -72,18 +73,18 @@ func newConsistentHashingAllocator(log logr.Logger) Allocator {
...
 		TargetsPerCollector.WithLabelValues(previousColName.String(), consistentHashingStrategyName).Set(float64(c.collectors[previousColName.String()].NumTargets))
 	}
-	colOwner := c.consistentHasher.LocateKey([]byte(target.Hash()))
-	targetItem := &TargetItem{
-		JobName:       target.JobName,
-		Link:          LinkJSON{Link: fmt.Sprintf("/jobs/%s/targets", url.QueryEscape(target.JobName))},
-		TargetURL:     target.TargetURL,
-		Label:         target.Label,
+	colOwner := c.consistentHasher.LocateKey([]byte(tg.Hash()))
+	targetItem := &target.Item{
```
Comment (jaronoff97) [inline cmd/otel-allocator/allocation/consistent_hashing.go:83]: nit: can we use a target constructor to make this `target.New`

File: cmd/otel-allocator/allocation/least_weighted.go [hunk 2]
```diff
@@ -94,13 +95,13 @@ func (allocator *leastWeightedAllocator) findNextCollector() *Collector {
...
 // INVARIANT: allocator.collectors must have at least 1 collector set.
-func (allocator *leastWeightedAllocator) addTargetToTargetItems(target *TargetItem) {
+func (allocator *leastWeightedAllocator) addTargetToTargetItems(tg *target.Item) {
 	chosenCollector := allocator.findNextCollector()
-	targetItem := &TargetItem{
-		JobName:       target.JobName,
-		Link:          LinkJSON{Link: fmt.Sprintf("/jobs/%s/targets", url.QueryEscape(target.JobName))},
-		TargetURL:     target.TargetURL,
-		Label:         target.Label,
+	targetItem := &target.Item{
```
Comment (jaronoff97) [inline cmd/otel-allocator/allocation/least_weighted.go:100]: nit: use constructor here too

**Review by jaronoff97 — COMMENTED** [review]
Just one small change to use the constructor, otherwise, looks great!

File: cmd/otel-allocator/allocation/consistent_hashing.go (hunk 1 above)
Comment (moh-osman3) [reply cmd/otel-allocator/allocation/consistent_hashing.go]: Thanks for the review! This is a good point, changed!

File: cmd/otel-allocator/allocation/least_weighted.go (hunk 2 above)
Comment (moh-osman3) [reply cmd/otel-allocator/allocation/least_weighted.go]: Fixed!

**Review by kristinapathak — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**